### PR TITLE
Make template errors into diagnostics

### LINKED ISSUE #38155: `<div [class.x]=></div>` crashes the forked type checker
# 🐞 bug report

## 🌍  Your Environment

Angular CLI: 10.0.3
Angular: 10.0.4

## 🔬 Minimal Reproduction

```sh
npm i -g @angular/cli
ng new ng10-crash
cd ng10-crash
ng serve
```

Then replace the app.component.html with `<div [class.x]=></div>` and see this in the terminal:

image

If you don't want the typechecker now slowing down the watcher in the main thread, you have to restart it.

## Expected behavior

typechecker remaining forked, even when there was an error

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

**Review by JoostK — COMMENTED** [review]
This accidentally fixes #38155 🎉

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by atscott — APPROVED** [review]
Reviewed-for: public-api

**Review by atscott — APPROVED** [review]

File: packages/compiler-cli/src/ngtsc/typecheck/diagnostics/src/id.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+interface HasTemplateId {
+  [TEMPLATE_ID]: TemplateId;
+}
+
+interface HasNextTemplateId {
+  [NEXT_TEMPLATE_ID]: number;
+}
+
+export function getTemplateId(clazz: ts.Declaration): TemplateId {
```
Comment (petebacondarwin) [inline packages/compiler-cli/src/ngtsc/typecheck/diagnostics/src/id.ts:25]: NIT, I think that this signature could be

```ts
export function getTemplateId(clazz: ts.Declaration & Partial<HasTemplate>): TemplateId {
```

in which case the cast and assignment are not needed. But perhaps you did it this way to keep the `HasTemplate` interface private?

**Review by petebacondarwin — APPROVED** [review]
LGTM both in general and
Reviewed-for: public-api

File: packages/compiler-cli/src/ngtsc/typecheck/diagnostics/src/id.ts (hunk 1 above)
Comment (alxhub) [reply packages/compiler-cli/src/ngtsc/typecheck/diagnostics/src/id.ts:25]: Yes, and it shouldn't be a part of the public API of `getTemplateId`. How the ID is stored is an internal matter.

**Pull request closed without merging**